### PR TITLE
Implement API Errors management with status code

### PR DESCRIPTION
**Pull request by rodanr**
### What does this PR contain ?
- Adds a better way to handle api errors with status code that can be beneficial for
  - Enables throwing API error with status code from service/repository layer without needing for gin context
  - Will help in efficient management of sentry logging, as the expected api error with project like validation errors, record not found doesn't needs to be debugged
- Tests for the utils.HandleError method

### CODE REVIEW

File: pkg/utils/handle_errors.go [hunk 1]
```diff
@@ -45,6 +42,23 @@ var sqlError *mysql.MySQLError
...
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		c.JSON(http.StatusNotFound, gin.H{
+			"error": err.Error(),
+		})
+		return
+	}
+
 	c.JSON(http.StatusInternalServerError, gin.H{
 		"error": err.Error(),
 	})
```
Comment (dipeshdulal) [inline pkg/utils/handle_errors.go]: @rodanr 
Need to send generic error message in case the error is not handled properly. 
Sending the same error message as returned from db or other services has security risks.

Comment (rodanr) [reply pkg/utils/handle_errors.go]: Thank you @dipeshdulal sir for you suggestion. Suggested changes have been made. Thank you

File: pkg/utils/handle_errors.go [hunk 2]
```diff
@@ -51,16 +53,16 @@ func HandleError(logger framework.Logger, c *gin.Context, err error) {
...
 
-	// will not captured by sentry if its an record not found error from gorm
+	// will not be captured by sentry if its a gorm.ErrRecordNotFound
 	if errors.Is(err, gorm.ErrRecordNotFound) {
 		c.JSON(http.StatusNotFound, gin.H{
-			"error": err.Error(),
+			"error": msgForUnhandledError,
 		})
 		return
 	}
```
Comment (dipeshdulal) [inline pkg/utils/handle_errors.go]: I think returning `api_errors.ErrRecordNotFound` is fine for this case instead of generic unhandled error. :)

File: pkg/utils/handle_errors.go [hunk 3]
```diff
@@ -28,10 +28,7 @@ func HandleErrorWithStatus(logger framework.Logger, c *gin.Context, statusCode i
 }
 
 // list static errors to filter
-var exceptStaticError = []error{
-	gorm.ErrRecordNotFound,
-	api_errors.ErrInvalidUUID,
-}
```
Comment (dipeshdulal) [inline pkg/utils/handle_errors.go:34]: @rodanr 
It's better to revert this change. 
We don't want to send these to sentry as these errors are not unexpected IMO. 
What do you think?

Comment (rodanr) [reply pkg/utils/handle_errors.go:34]: `gorm.ErrRecordNotFound` is handled manually to have a better status code of 404 and has skipped sentry logging too
	if errors.Is(err, gorm.ErrRecordNotFound) {
		c.JSON(http.StatusNotFound, gin.H{
			"error": gorm.ErrRecordNotFound.Error(),
		})
		return
	}
	
	api_errors.ErrInvalidUUID will be automatically be prevented from sentry logging as it is of new APIError type, for whose we are preventing sentry logging for its all instances as this are predicted error from user behaviour and doesnt need to be taken care of

**Review by dipeshdulal — APPROVED** [review]
@rodanr 
Thank you for your work. 🎖️ 
Merging this changes for now. 

There are changes necessary such as; 
- Implementing Ginkgo+Mockery as test suite instead of current implementation. 
- Improving sentry initialization and unit testing sentry captures. 

Will will be tackled later with collaboration with unit test integration team. 🙏

**Comment by rodanr** [comment]
> @rodanr Thank you for your work. 🎖️ Merging this changes for now.
> 
> There are changes necessary such as;
> 
> * Implementing Ginkgo+Mockery as test suite instead of current implementation.
> * Improving sentry initialization and unit testing sentry captures.
> 
> Will will be tackled later with collaboration with unit test integration team. 🙏

Thank you @dipeshdulal sir. Will work on that.
